### ISSUE TITLE
#[no_mangle] is not denied on EII definitions

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust#159015]
I tried this code:

```rust
#![feature(extern_item_impls)]

#[eii]
fn foo();

#[foo]
#[unsafe(no_mangle)]
fn bar() {}
```

I expected to see this happen: Error emitted as EIIs work through matching symbol names.

Instead, this happened: No error is emitted.

### Meta

`rustc --version --verbose`:
```
1.99.0-nightly (2026-07-08 14cae681329a63c622a6)
```
